client/my-sites/importer/site-importer/url-validation.js

```diff
@@ -0,0 +1,45 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import url from 'url';
+import { isWebUri } from 'valid-url';
+import { translate } from 'i18n-calypso';
```

Comment:
Is this a valid way to use translate function (outside of React components)?

Reply:
Sure thing!
IIRC, the only gotcha with using translate directly is when it's called before the translations are properly loaded - in which case I think the base string would be returned.

This might happen if, say, we called translate in the files global scope, outside of functions etc:

```js
import { translate } from 'i18n-calypso';
import { inRange, trim } from 'lodash';

const someMessage = translate( 'some message' );
```

But because here we're calling translate from a function that'll be called by a component, we can safely assume that the translations are loaded at the time of calling `translate` 👌

Reply:
👍